python/ray/train/context.py

```diff
@@ -84,6 +91,22 @@ def get_context() -> TrainContext:
 
     See the :class:`~ray.train.TrainContext` API reference to see available methods.
     """
+    from ray.train._internal.session import get_session
+    from ray.tune import get_context as get_tune_context
+
+    # If we are running in a Tune function, switch to Tune context.
+    if get_session() and get_session().world_rank is None:
+        warnings.warn(
+            (
+                "`ray.train.get_context()` should be switched to "
+                "`ray.tune.get_context()` when running in a function "
+                "passed to Ray Tune. This will be an error in the future."
+            ),
+            RayDeprecationWarning,
+            stacklevel=2,
+        )
+        return get_tune_context()
```

Comment:
Example of calling `ray.train.get_context` in Ray Tune routing to `ray.tune.get_context` with a warning.